pkg/helm/client.go

```diff
@@ -76,36 +41,44 @@ func newClient(opts HelmOptions) (*HelmClient, error) {
 		}
 		kversion = sv
 	}
-	regcli, err := registry.NewClient(registryOpts...)
-	if err != nil {
-		return nil, fmt.Errorf("create registry client: %w", err)
-	}
-	if opts.RESTClientGetter == nil {
-		cfgFlags := &genericclioptions.ConfigFlags{}
-		if opts.KubeConfig != "" {
-			cfgFlags.KubeConfig = &opts.KubeConfig
-		}
-		opts.RESTClientGetter = cfgFlags
-	}
+
 	return &HelmClient{
-		tmpdir:           tmpdir,
-		kversion:         kversion,
-		restClientGetter: opts.RESTClientGetter,
-		regcli:           regcli,
-		logFn:            opts.LogFn,
-		airgapPath:       opts.AirgapPath,
+		helmPath:              helmPath,
+		executor:              newBinaryExecutor(helmPath, opts.LogFn),
+		tmpdir:                tmpdir,
+		kversion:              kversion,
+		kubernetesEnvSettings: opts.KubernetesEnvSettings,
+		airgapPath:            opts.AirgapPath,
+		repositories:          []*repo.Entry{},
 	}, nil
 }
 
+func getHelmPath(opts HelmOptions) (string, error) {
+	if opts.HelmPath != "" {
+		// used in operations that do not have the helm binary embedded (i.e. upgrade via operator, etc..)
+		return opts.HelmPath, nil
+	}
+	if hp := os.Getenv("HELM_BINARY_PATH"); hp != "" {
+		// used in tests so that we don't pass the helm binary path all over the place
+		return hp, nil
+	}
+	hp, err := goods.Binary("helm")
```

Comment:
would it be possible to pass around a HelmClient rather than an EnvSettings so you dont have to use the goods package outside of the cli? Looks like you would have to factor out logging into the call to the client functions.

Reply:
hmm this will make it hard to override the log function for the helm client and other operation specific fields.

Reply:
what if i pass the helm binary path explicitly from callers instead of using the goods package?

Reply:
If you didnt want to add the log as an argument to all methods, could you add the log to the context? Similar to this https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/log